### DISCUSSION TITLE
Getting `Access to requested resource is denied` error

### DISCUSSION THREAD

**Discussion by saleweaver** [saleweaver/python-amazon-sp-api#36] [FAQ] [answered]
I get an error saying:

```
SellingApiForbiddenException: [{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': 'Access token is missing in the request header.'}]
```

**Comment by saleweaver** [comment, marked as answer]
This is most likely due to one of below credentials being wrong:

```
  refresh_token
  lwa_app_id
  lwa_client_secret
```

**Comment by t-wittkopf** [comment]
The `Access to requested resource is denied` error can also be a result of using the wrong marketplace in your client.
Your app is registered to a certain marketplace. You may have to adjust the marketplace value in your client to correspond with your app.
The clients default marketplace is US:
`marketplace: Marketplaces = Marketplaces.US`

**Reply by sdaeric19** [reply]
@t-wittkopf Where do we find out what marketplace our app is registered to?  My store is based in UK.

When I try below block of code
```
from sp_api.api import Orders
from sp_api.base import SellingApiException, Marketplaces
from datetime import datetime, timedelta

# orders API
try:
    res = Orders(marketplace=Marketplaces.GB).get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=1)).isoformat())
    print(res.payload)  # json data
except SellingApiException as ex:
    print(ex)
```
My response is `[{'message': 'We encountered an internal error. Please try again.', 'code': 'InternalFailure'}]`

**Reply by saleweaver** [reply]
Are you running this against the SANDBOX?
I've never seen this error in production (if I remember correctly)

**Reply by sdaeric19** [reply]
Yup that was it. I just took out the "SANDBOX" envt variable and orders are coming through now. Oh, the joy of seeing an API Response!

Thanks for your help and support hugely appreciate!

**Reply by dehidehidehi** [reply]
> The `Access to requested resource is denied` error can also be a result of using the wrong marketplace in your client.
> Your app is registered to a certain marketplace. You may have to adjust the marketplace value in your client to correspond with your app.
> The clients default marketplace is US:
> `marketplace: Marketplaces = Marketplaces.US`

This fixed my issue, thank you.

**Comment by elburro1887** [comment]
I just got bit by this issue, and the reason was the wrong marketplace was set.

Maybe it would be good to add this info to the main README.md at

https://github.com/saleweaver/python-amazon-sp-api/blob/master/README.md#usage

Thanks!
